### PR TITLE
[SHLWAPI][INCLUDE] Initial implement SHGetAppCompatFlags

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Half-implement `shlwapi!SHGetAppCompatFlags` function.

JIRA issue: N/A

### CODE REVIEW

**Comment by learn-more** [comment]
Do you need this for a specific application?

If not, it might be nice to at least implement this behavior:
> The filename flags are evaluated just once per process, when this function is first called with any of these flags in the given mask. If the process is marked (in its executable header) as requiring Windows 5.00 or higher, then none of these compatibilty flags apply to it and no identification is attempted.

**Comment by katahiromz** [comment]
`shell32!ShellExecuteA` is probably calling `SHGetAppCompatFlags`. This function is provided for backward compatibility of old programs.

File: sdk/include/reactos/shlwapi_undoc.h [hunk 1]
```diff
@@ -119,6 +119,33 @@ INT WINAPI SHUnicodeToAnsi(LPCWSTR lpSrcStr, LPSTR lpDstStr, INT iLen);
...
+#define LOADCOLUMNHANDLER 0x00000040
+#define ANSI 0x00000080
+#define WIN95SHLEXEC 0x00000200
+#define STAROFFICE5PRINTER 0x00000400
+#define NOVALIDATEFSIDS 0x00000800
+#define FILEOPENNEEDSEXT 0x00001000
+#define WIN95BINDTOOBJECT 0x00002000
+#define IGNOREENUMRESET 0x00004000
+#define ANSIDISPLAYNAMES 0x00010000
+#define FILEOPENBOGUSCTRLID 0x00020000
```
Comment (learn-more) [inline sdk/include/reactos/shlwapi_undoc.h:144]: FILEOPEN could refer to the FileOpen dialog from comdlg32

File: sdk/include/reactos/shlwapi_undoc.h [hunk 2]
```diff
@@ -119,6 +119,33 @@ INT WINAPI SHUnicodeToAnsi(LPCWSTR lpSrcStr, LPSTR lpDstStr, INT iLen);
...
+#define FLUSHNOWAITALWAYS 0x00000001
+#define DOCOBJECT 0x00000002
+#define CORELINTERNETENUM 0x00000004
+#define OLDCREATEVIEWWND 0x00000004
+#define WIN95DEFVIEW 0x00000004
+#define MYCOMPUTERFIRST 0x00000008
+#define OLDREGITEMGDN 0x00000010
+#define LOADCOLUMNHANDLER 0x00000040
+#define ANSI 0x00000080
+#define WIN95SHLEXEC 0x00000200
```
Comment (learn-more) [inline sdk/include/reactos/shlwapi_undoc.h:137]: This one is most likely for ShellExecute, yes

File: sdk/include/reactos/shlwapi_undoc.h [hunk 3]
```diff
@@ -119,6 +119,33 @@ INT WINAPI SHUnicodeToAnsi(LPCWSTR lpSrcStr, LPSTR lpDstStr, INT iLen);
...
+/*
+ * Flags for SHGetAppCompatFlags.
+ * See https://www.geoffchappell.com/studies/windows/shell/shlwapi/api/util/getappcompatflags.htm
+ */
+#define CONTEXTMENU 0x00000001
+#define FLUSHNOWAITALWAYS 0x00000001
+#define DOCOBJECT 0x00000002
+#define CORELINTERNETENUM 0x00000004
+#define OLDCREATEVIEWWND 0x00000004
+#define WIN95DEFVIEW 0x00000004
```
Comment (learn-more) [inline sdk/include/reactos/shlwapi_undoc.h:132]: Probably for CDefView?

**Comment by katahiromz** [comment]
Counterproductive. Postponed.

**Pull request closed without merging**